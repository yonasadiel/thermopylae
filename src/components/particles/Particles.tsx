import { useEffect } from 'react';
import { useSettings } from '../../hooks';

const Particles = () => {
    const { settings } = useSettings();
    useEffect(() => {
        window.particlesJS('particles-js', particlesConfig, function () {
            // callback
        });
    }, [settings.themeForegroundColor]);
    return (
        <div
            id="particles-js"
            style={{
                display: settings.themeBackgroundParticlesEnabled ? 'block' : 'none',
                zIndex: 0,
            }}
        ></div>
    );
}

export default Particles;

const particlesConfig = {
    "particles": {
        "number": {
            "value": 120,
            "density": {
                "enable": true,
                "value_area": 800
            }
        },
        "color": {
            "value": "#ffffff"
        },
        "shape": {
            "type": "circle",
            "stroke": {
                "width": 0,
                "color": "#000000"
            },
            "polygon": {
                "nb_sides": 5
            },
            "image": {
                "src": "img/github.svg",
                "width": 100,
                "height": 100
            }
        },
        "opacity": {
            "value": 0.2,
            "random": false,
            "anim": {
                "enable": false,
                "speed": 1,
                "opacity_min": 0.1,
                "sync": false
            }
        },
        "size": {
            "value": 3,
            "random": true,
            "anim": {
                "enable": false,
                "speed": 12.181158184520175,
                "size_min": 0.1,
                "sync": false
            }
        },
        "line_linked": {
            "enable": true,
            "distance": 160.3412060865523,
            "color": "#ffffff",
            "opacity": 0.2725800503471389,
            "width": 0.8017060304327615
        },
        "move": {
            "enable": true,
            "speed": 0.4,
            "direction": "top",
            "random": true,
            "straight": false,
            "out_mode": "out",
            "bounce": false,
            "attract": {
                "enable": false,
                "rotateX": 600,
                "rotateY": 1200
            }
        }
    },
    "interactivity": {
        "detect_on": "canvas",
        "events": {
            "onhover": {
                "enable": true,
                "mode": "bubble"
            },
            "onclick": {
                "enable": true,
                "mode": "repulse"
            },
            "resize": true
        },
        "modes": {
            "grab": {
                "distance": 400,
                "line_linked": {
                    "opacity": 1
                }
            },
            "bubble": {
                "distance": 100,
                "size": 2,
                "duration": 2,
                "opacity": 1,
                "speed": 3
            },
            "repulse": {
                "distance": 200,
                "duration": 0.4
            },
            "push": {
                "particles_nb": 4
            },
            "remove": {
                "particles_nb": 2
            }
        }
    },
    "retina_detect": false
}
