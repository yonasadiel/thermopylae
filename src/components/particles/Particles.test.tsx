import { test, expect, describe } from '@jest/globals';
import { render } from '@testing-library/react';
import Particles from './Particles';
import { fn } from 'jest-mock';

describe('Particles', () => {
    test('should render correctly', () => {
        window.particlesJS = fn();
        const result = render(<Particles />);
        expect(result.baseElement).toMatchSnapshot();
    });
});