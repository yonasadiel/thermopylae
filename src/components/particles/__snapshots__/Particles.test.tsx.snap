// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Particles should render correctly 1`] = `
<body>
  <div>
    <div
      id="particles-js"
      style="display: block; z-index: 0;"
    />
  </div>
</body>
`;
